### PR TITLE
Sumida c75

### PR DESCRIPTION
**Pull request by drid**
Sumida c75 SMD Inductor

image

Datasheet http://products.sumida.com/products/pdf/CR75.pdf

---

All contributions to the kicad library must follow the [KiCad library convention](http://kicad-pcb.org/libraries/klc/)

Thanks for creating a pull request to contribute to the KiCad libraries! To speed up integration of your PR, please check the following items:

- [x] Provide a URL to a datasheet for the footprint(s) you are contributing
- [x] An example screenshot image is very helpful 
- [x] If there are matching symbol or 3D model pull requests, provide link(s) as appropriate
- [x] Check the output of the Travis automated check scripts - fix any errors as required
- [x] Give a reason behind any intentional library convention rule violation.

### CODE REVIEW

**Comment by chschlue** [comment]
You accidentally pushed https://github.com/KiCad/kicad-footprints/pull/1771 here as well.

**Comment by drid** [comment]
> You accidentally pushed #1771 here as well.

Fixed

**Comment by gmsotavio** [comment]
Hello,

I believe which you footprint is violation KLC F4.2 (http://kicad-pcb.org/libraries/klc/F4.2/)

"Pin 1 should be located at the top left "
image

**Comment by drid** [comment]
> Hello,
> 
> My first thoughts about this contribution:
> 
> Check the following KLC violations:
> 
> * KLC F4.2 (http://kicad-pcb.org/libraries/klc/F4.2/)
> 
> "Pin 1 should be located at the top left "
> image
> 
> * KLC F5.2 (http://kicad-pcb.org/libraries/klc/F5.2/)
>   "RefDes must be centered on component body (inside component outline)"
>   "Orientation of RefDes should match major component axis"

Thank you for your feedback, I corrected it as you suggested

**Comment by gmsotavio** [comment]
Thanks!

There are more three things that could be changed in order to be compliant with KLC.
- When you rotated the entire body of the footprint, you forgot to change the orientation of the pads. The rotation should be 0 degrees. To fix this, change the pad rotation to get 0 degrees, and then, swap x and y dimensions.
- Your Courtyard is beneath of Fab layer (Your courtyard is the pink line in the image bellow). 
image
Actually, it must follows the KLC F5.3 (http://kicad-pcb.org/libraries/klc/F5.3/). To fix this, move the Courtyard lines in order to get 0.25mm of clearance from pads and Fab layer. Remember which all courtyard line elements are placed on a 0.01mm grid. Take a look in the example bellow
image

- Would be nice if you provided a better description of your footprint instead only the datasheet link. See some examples found in the current KiCad library
image

The remaining dimensions seems OK to me.
image

File: Inductor_SMD.pretty/L_Sumida_CR75.kicad_mod [hunk 1]
```diff
@@ -1,30 +1,30 @@
-(module L_Sumida_CR75 (layer F.Cu) (tedit 5D4EE90A)
+(module L_Sumida_CR75 (layer F.Cu) (tedit 5D4F3CE3)
   (descr http://products.sumida.com/products/pdf/CR75.pdf)
   (tags "Inductor SMD")
   (attr smd)
-  (fp_text reference REF** (at 5.08 -2.667 90) (layer F.SilkS)
+  (fp_text reference REF** (at 0.04 -4.48 180) (layer F.SilkS)
```
Comment (gmsotavio) [inline Inductor_SMD.pretty/L_Sumida_CR75.kicad_mod]: x coordinate must be 0.

File: Inductor_SMD.pretty/L_Sumida_CR75.kicad_mod [hunk 2]
```diff
@@ -1,30 +1,30 @@
...
     (effects (font (size 1 1) (thickness 0.15)))
   )
-  (fp_text value L_Sumida_CR75 (at 0 -5.1) (layer F.Fab)
+  (fp_text value L_Sumida_CR75 (at 0.04 4.52) (layer F.Fab)
```
Comment (gmsotavio) [inline Inductor_SMD.pretty/L_Sumida_CR75.kicad_mod]: x coordinate must be 0.

File: Inductor_SMD.pretty/L_Sumida_CR75.kicad_mod [hunk 3]
```diff
@@ -1,30 +1,30 @@
...
     (effects (font (size 1 1) (thickness 0.12)))
   )
-  (fp_text user %R (at 0 5.6) (layer F.Fab)
+  (fp_text user %R (at 0.04 -0.08) (layer F.Fab)
```
Comment (gmsotavio) [inline Inductor_SMD.pretty/L_Sumida_CR75.kicad_mod]: x coordinate must be 0.

File: Inductor_SMD.pretty/L_Sumida_CR75.kicad_mod [hunk 4]
```diff
@@ -1,30 +1,30 @@
...
+  (fp_line (start -3.9 3.5) (end 3.9 3.5) (layer F.Fab) (width 0.1))
+  (fp_line (start 4.2 3.8) (end -4.2 3.8) (layer F.SilkS) (width 0.12))
+  (fp_line (start 4.2 -3.8) (end 4.2 3.8) (layer F.SilkS) (width 0.12))
+  (fp_line (start -4.2 -3.8) (end 4.2 -3.8) (layer F.SilkS) (width 0.12))
+  (fp_line (start -4.2 3.8) (end -4.2 -3.8) (layer F.SilkS) (width 0.12))
+  (fp_line (start -3.9 -3.5) (end -3.9 3.5) (layer F.CrtYd) (width 0.05))
+  (fp_line (start 3.9 -3.5) (end -3.9 -3.5) (layer F.CrtYd) (width 0.05))
+  (fp_line (start 3.9 3.5) (end 3.9 -3.5) (layer F.CrtYd) (width 0.05))
+  (fp_line (start -3.9 3.5) (end 3.9 3.5) (layer F.CrtYd) (width 0.05))
+  (pad 2 smd roundrect (at 2.5 -0 90) (size 5.5 3) (layers F.Cu F.Paste F.Mask))
```
Comment (gmsotavio) [inline Inductor_SMD.pretty/L_Sumida_CR75.kicad_mod]: Pad rotation must be 0 degrees.

File: Inductor_SMD.pretty/L_Sumida_CR75.kicad_mod [hunk 5]
```diff
@@ -1,30 +1,30 @@
...
+  (pad 1 smd roundrect (at -2.5 -0 90) (size 5.5 3) (layers F.Cu F.Paste F.Mask))
```
Comment (gmsotavio) [inline Inductor_SMD.pretty/L_Sumida_CR75.kicad_mod]: Pad rotation must be 0 degrees.

**Comment by gmsotavio** [comment]
Thanks for the changes.
image
image

**Comment by myfreescalewebpage** [comment]
Hello @drid just looked this PR, one comment:

- [ ] F.SilkS offset from pads/F.Fab lines should be 0.11mm, it's currently too large

Else, the footprint is ok.

Joel

**Comment by drid** [comment]
image
image

**Comment by myfreescalewebpage** [comment]
Not sure my comment was properly written and well understood: The clearance between F.Fab and F.SilkS should be 0.11mm but the F.SilkS clearance with all pad should be at least 0.2mm (https://kicad-pcb.org/libraries/klc/F5.1/)

Currently on top and bottom it's too large and on left and right it's too closed to the pad.

Joel

**Comment by drid** [comment]
Adjusted silkscreen based on component size and pad clearance then adjusted F.Fab for 0.11 clearance (is there a rule for 0.11?)

**Comment by chschlue** [comment]
One rule is that it has to be at least the line width from exposed copper, preferably 0.2mm (line edge to pad edge).
Another rule is that no silk must be beneath SMT components.

Also, outline silk is supposed to represent the part's outline, i.e. it must not be too far.
The FP generators position silk 0.11mm from fab (center to center) because that's 0.5 silk line width + 0.5 fab line width, meaning the lines touch and look neat that way.
0.11mm isn't set in stone, though.

**Pull request merged**